packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -411,7 +411,6 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
         }
         this.toDisposeOnConnect.dispose();
         this.toDispose.push(this.toDisposeOnConnect);
-        this.term.reset();
```

Comment:
I think it was important for some reasons,  like you please track back while it was added. We have to test for regressions.

Reply:
yes it is important. in my pr it is moved to this place https://github.com/eclipse-theia/theia/pull/7260/files#diff-38e65dfb04ff1bf292adf8abf4e1b9d5R99

Reply:
but it is not about tasks, it's about resetting a state for terminal generally on reconnection

I think it has to stay here.

Reply:
Removing it will cause https://github.com/eclipse-theia/theia/issues/2554 again.

Reply:
hmm... please could you advise on how to resolve this, as i am stuck:

1.  `this.term.reset()` clears all output in terminal widget.
2. terminal should have no knowledge of tasks. 
3. we want to retain the existing terminal output when a task is started

so if i put `this.term.reset()` back, i wouldn't be able to achieve # 3 due to the reason described in # 2 .

Reply:
> `this.term.reset()` clears all output in terminal widget.

It does not clear the output, it resets it from the shell. If the shell is still have output it should be there. What kind of output you mean?

Reply:
I take a closer look at the behavior:

![Peek 2020-03-13 07-51](https://user-images.githubusercontent.com/37082801/76618727-a2439e80-64ff-11ea-892f-a0de67b25fd6.gif)

Reply:
maybe i am wrong. looks like `term.reset()` does not clear the output directly. it is something else that clears the terminal output after the `term.reset()`.

Reply:
ok i believe `terminal.reset()` clears the  terminal itself:

reset() prints '\x1bc':
https://github.com/xtermjs/xterm.js/blob/master/typings/xterm.d.ts#L1007

and '\x1bc' clears the terminal:
https://askubuntu.com/a/616773

and i personally don't think `term.reset()` belongs to `terminal.connectTerminalProcess()`.

in case that you disagree with me, i will find a work-around.

Reply:
@akosyakov 
I was trying to reproduce #2554 without `term.reset()` and I wasn't able to (please see GIF below). Please kindly let me know if i was reproducing it the correct way. Thank you.

![Peek 2020-03-15 19-02](https://user-images.githubusercontent.com/37082801/76712456-b221d500-66ef-11ea-8a77-38f9c04623c3.gif)

Reply:
I was trying to reproduce it with your PR as well and could not, something seems to change in the meantime. Let's try without it. 👍